### PR TITLE
SAK-44265 Moved bullhorns to use Server Sent Events

### PR DESCRIPTION
**Pull request by adrianfish**
https://jira.sakaiproject.org/browse/SAK-44265

### CODE REVIEW

**Comment by mpellicer** [comment]
Wohooo, this sounds amazing!

File: assignment/api/src/java/org/sakaiproject/assignment/api/AddAssignmentBullhornHandler.java [hunk 1]
```diff
@@ -150,8 +146,6 @@ protected void doInTransactionWithoutResult(TransactionStatus status) {
             }
         });
 
-        users.forEach(u -> countCache.remove(u));
```
Comment (ottenhoff) [inline assignment/api/src/java/org/sakaiproject/assignment/api/AddAssignmentBullhornHandler.java:153]: what's the logic around removing this cache?

File: deploy/pom.xml [hunk 2]
```diff
@@ -392,6 +392,22 @@
...
+        <dependency>
+            <groupId>org.reactivestreams</groupId>
+            <artifactId>reactive-streams</artifactId>
+            <version>1.0.3</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.projectreactor</groupId>
+            <artifactId>reactor-core</artifactId>
+            <version>3.3.9.RELEASE</version>
```
Comment (ottenhoff) [inline deploy/pom.xml:404]:
```suggestion
            <version>3.3.15.RELEASE</version>
```

File: lessonbuilder/api/src/java/org/sakaiproject/lessonbuildertool/api/AddLessonsCommentBullhornHandler.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                        done.add(to);
+                    }
+                }
+
+                // Get all the comments in the same item
+                List<SimplePageComment> comments
+                    = simplePageToolDao.findCommentsOnItems(
+                        Arrays.asList(new Long[] {comment.getItemId()}));
+
+                if (comments.size() > 1) {
```
Comment (ottenhoff) [inline lessonbuilder/api/src/java/org/sakaiproject/lessonbuildertool/api/AddLessonsCommentBullhornHandler.java:84]:
```suggestion
                if (comments != null && comments.size() > 1) {
```

File: assignment/api/src/java/org/sakaiproject/assignment/api/AddAssignmentBullhornHandler.java (hunk 1 above)
Comment (adrianfish) [reply assignment/api/src/java/org/sakaiproject/assignment/api/AddAssignmentBullhornHandler.java:153]: We don't maintain counts server side any longer. The clientside stuff just counts them as they come in.

Comment (adrianfish) [reply assignment/api/src/java/org/sakaiproject/assignment/api/AddAssignmentBullhornHandler.java:153]: In the entityprovider, there used to be a call just to get the counts - that was what got short polled every 5 seconds. That call is no longer happening, so no cache needed.

**Review by ottenhoff — APPROVED** [review]

**Pull request merged**